v1.0.2 (#5)
api3dao/api3-whitepaper: api3-whitepaper.tex

First-party oracles are integral to the API3 solution.
This means each API is served by an oracle that is operated by the entity that owns the API, rather than a third-party.

> In this section, we will discuss the benefits of using first-party oracles, and how we made it feasible for API providers to operate their own oracles with Airnode.

Reviewer: ```suggestion
In this section, we will discuss the benefits of using first-party oracles and how API3 makes it feasible for API providers to operate their own oracles with Airnode.
```

Existing solutions using third-party oracles depend on over-redundancy at the oracle level to cover for this, which results in excessive costs as mentioned in Section~\ref{sec:ineffective-redundancy}.
API3 envisions each API to only be served by its first-party oracle, which means the redundancy has to be implemented at the level of the individual Airnode.
The node being fully-serverless enables this to be done easily across different availability zones of a single cloud provider, or even across multiple cloud providers.
It should also be mentioned that it is possible to operate the containerized version of Airnode on-premises, yet using the serverless version will be recommended for almost all use cases.

> Airnode is open-sourced with MIT License and ready to enable first-party oracles\footnote{\url{https://github.com/api3dao/airnode}}.

Reviewer: ```suggestion
Airnode is open-sourced, MIT Licensed, and ready to enable first-party oracles\footnote{\url{https://github.com/api3dao/airnode}}.
```

Author: It's the specific license that determines what we mean by open source, so I prefer "open-sourced with *" here

Reviewer: Sounds good 👍

There is a chicken-and-egg problem when it comes to integrating APIs to oracles.
If there is no existing demand for an API in an oracle ecosystem, nobody is incentivized to do the integration.
If the API is not available due to a lack of integration, nobody develops applications that will create the demand.
This was identified as a key friction point for the Chainlink ecosystem, and Honeycomb API Marketplace was proposed as a solution~\cite{benligiray:2019}.

> Honeycomb had integrated a large number of premium APIs to Chainlink oracles, and as a result, this marketplace served an API variety that is unmatched in any oracle ecosystem of its time.

Reviewer: ```suggestion
Honeycomb had integrated a large number of premium APIs to Chainlink oracles, and as a result, this marketplace served an API variety that was unmatched in any oracle ecosystem at the time.
```

\subsection{API3 DAO}
\label{sec:api3-dao}

To decentralize the governance of both dAPIs and the project as a whole, API3 is governed by a DAO\footnote{\url{https://api3.eth.link} is a dashboard for the API3 DAO that is maintained by the core technical team.}.
The governance is entirely decentralized and open, meaning that all stakeholders are able to participate in the governance of the project directly.
This is achieved through the API3 token, which grants voting power in the API3 DAO through the mechanics described in Section~\ref{sec:api3-tokenomics}.

The DAO votes on high-level parameters regarding mechanics such as staking incentives and collateralization.

> Additionally, the DAO gives out grants and by consequence decide on the general direction of the project.

Reviewer: ```suggestion
Additionally, the DAO awards grants and by consequence decides on the general direction of the project.
```

\subsection{dAPI monetization and API provider compensation}
\label{sec:dapi-monetization-and-api-provider-compensation}

API subscription fees are commonly paid monthly or annually, as this scheme suits both API providers and their clients.
API3 will aim to follow the same scheme for dAPIs.
To gain access to a dAPI, the user will pay a recurring subscription fee, which may either be fixed or customized for the user based on the specific use case.
These prices will be determined by the respective team, and will include the premium if the user wants to receive the insurance service described in Section~\ref{sec:quantifiable-security-through-insurance}.
The payment will be able to be made in any cryptocurrency, which will be received by the DAO in API3 tokens through a liquidity pool-based decentralized exchange.

API providers will be compensated periodically at fixed rates, which will fit their existing pricing models.
This will be done using stablecoins wherever possible, yet as mentioned in Section~\ref{sec:barriers-to-api-providers-operating-oracles}, some API providers categorically reject handling cryptocurrency as payment.

> In such cases, the DAO will provide a grant that will be paid out in return of the proof that the API provider is compensated in fiat by the grantee.

Author: This is mostly about dAPI revenue, which is covered in the staking section. We can revive the content (especially the part about API provider compensation) as it gets fleshed out.

These cross-platform integrations will be implemented and maintained through grants, which will be given to teams partly composed of parties from the respective smart contract platform ecosystems.

We have described a scheme for compensating API providers in fiat currency in Section~\ref{sec:dapi-monetization-and-api-provider-compensation}.
Regarding cross-platform dAPI subscription fees, we need the API3 DAO to be compensated across smart contract platforms, which is a similar problem.
Therefore, we will use an analog of the same solution, where a grantee will pay the API3 DAO in return for being authorized to receive payments on its behalf on another smart contract platform.
For example, if the DAO needs to be paid 100 API3 tokens as the subscription fee, the grantee will pay 90 API3 tokens to the DAO, which will result in the understanding that the grantee is authorized to receive the 100 API3 tokens-worth of payment on another chain.
This process will be formalized as required.

By implementing the data bridge and the payment channel, API3 will be able to serve other smart contract platforms without requiring them to interact with Ethereum or handle API3 tokens.

> Note that Ethereum will be used as the dispute resolution layer for the insurance service described in Section~\ref{sec:quantifiable-security-through-insurance} until a cross-platform alternative emerges.

Author: This foresees bridging of data between chains, which we moved away from as early as 2020. We are working on a mechanism for cross-chain monetization, so the bridging of payments is also irrelevant. Again, this can be revived after it gets implemented.

> \subsection{API3 tokenomics}

Author: I had linked the medium post that explains the new scheme but the majority of the people were understandably still referring to the inflation schedule graphs. So I revised this entire section based on the medium post.

>     \item Staking: Grants inflationary rewards, which are balanced by deflationary mechanics such as tokens being burned or time-locked to receive API3 services.

Reviewer: ```suggestion
    \item Staking: Grants inflationary rewards, which are balanced by deflationary mechanics such as burning or time-locking tokens in exchange for API3 services.
```

The miners that validate the correct operation of these applications receive the entirety of the transaction fees paid by their users.

> In our case, API3 stakers are the participants of the consensus mechanism that governs the project, which implies that they should be the sole beneficiary of dAPI monetization.

Reviewer: Wouldn't API providers contributing to the dAPI in a way also be beneficiaries? I might remove "sole".

Author: API3 would have dAPI operating costs in the form of API provider compensation, gas fees, etc. Then, it (indirectly) receives what is left of this, i.e. the profit. So by the sole beneficiary, I mean the sole recipient of the operating profits. Though this may not be entirely true, because a non-staker will also benefit from the deflation. I'm 50/50 on this so just let me know what you think is the best idea based on this.

Reviewer: This explanation was helpful and now I don't think "sole" needs changing, but "primary" could be a nice replacement. I don't think "sole recipient of operating profits", while perhaps more descriptive, is necessary.

EIP-1559 posed that the primarily transaction fee-based miner incentivization scheme caused economic instability~\cite{roughgarden:2020}.
With the update, the users now pay a base fee that floats with network usage, and this fee gets burned.
Instead of the full transaction fee, the miners receive a stable, inflationary block reward.
(EIP-1559 also includes a priority fee mechanism that counters Ethereum-specific issues such as uncle blocks and the block size limit, which do not have a direct correspondence in our analogy.)

A DAO that distributes revenue to the stakers is similar to pre-EIP-1559 mining rewards in that it will cause instability.
Each revenue distribution event creates a discontinuous jump in terms of incentives, which will be abused for profit.
For example, a group of stakers may propose dAPI subscription fees to be paid yearly instead of monthly, and unstake as soon as the subscription fees are paid and the revenue is distributed.
In addition to punishing long term participants, this will cause the total staked amount to oscillate according to periodic subscription fee payments.
Just as the problem is an analog, so is the solution.

> The API3 DAO will require the users to burn or time-lock API3 tokens to receive services, effectively creating a deflationary mechanism that benefits the staking reward recipients indirectly.

Author: Part 1 - Why we only pay inflationary rewards and not distribute revenue

API3 services being used will cause a shortage in API3 supply, which will benefit all token holders, and not only the stakers.
This implies that the staking rewards should be regulated to ensure that a certain portion of the API3 token holders stake, which will secure the governance of the project and provide collateral for insurance products.
To this end, the API3 DAO sets a ``stake target", which is a percentage of the total token supply.

> Every week, the reward amount is updated iteratively, i.e., increases if the staked amount is below the stake target, and vice versa.\footnote{\url{https://enormous.cloud/dao/api3/tracker} is an explorer for the API3 DAO that is maintained by Enormous.Cloud. You can use it to see the history of the reward amounts to date.}

Author: Part 2 - How much staking reward is paid, why

> It is challenging to incentivize good governance through staking rewards, as it is not something that can be reliably detected.

Reviewer: ```suggestion
It is challenging to incentivize good governance through staking rewards, as governance is not easily quantified.
```

For example, participation is generally considered to be desirable, yet voting on a proposal that will already pass is redundant in most cases, and actively abstaining is a legitimate stance.
As a solution, we only reward long-term outcomes.

> Specifically, the staking rewards are paid out weekly, yet each payment becomes withdrawable after a full year.

Reviewer: ```suggestion
Specifically, the staking rewards are paid out weekly, yet each payment becomes withdrawable only after a full year.
```

> This incentivizes all stakers to cooperate to maximize the token price one year from now, and the rolling nature of this release prevents instability.

Author: Part 3 - When do the staking rewards become available, why

\subsubsection{Governance}
\label{sec:governance}

The only way to gain representation at the API3 DAO is to stake API3 tokens in the insurance collateral pool.
As such, the governing parties are exposed to all risks and rewards of API3, and will govern to optimize them.

Inflationary rewards and the staked governance tokens being used as collateral will create a positive feedback loop in terms of governance quality.

> Initial token holders will have to stake and expose themselves to risk if they do not want to lose value to inflation.

Reviewer: ```suggestion
Token holders will have to stake and expose themselves to risk if they do not want to lose value to inflation.
```

\newpage
\normalsize
\appendix
\section{Whitepaper Versions}
\label{sec:versions}

> \href{https://github.com/api3dao/api3-whitepaper/releases/tag/v1.0.2}{v1.0.2} -- Burak Benligiray, November 2021.

Author: Authors to be updated
